test/Macros/macro_plugin_basic.swift

```diff
@@ -3,11 +3,23 @@
 // RUN: %empty-directory(%t)
 // RUN: split-file %s %t
 
-// RUN: sed -i '' -e 's#PYTHON_EXEC_PATH#%{python}#' %t/plugin
-// RUN: sed -i '' -e 's#UTILS_DIR_PATH#%utils#' %t/plugin
-// RUN: chmod +x %t/plugin
+// RUN: %clang \
+// RUN:  -isysroot %sdk \
+// RUN:  -I %swift_src_root/include \
+// RUN:  -L %swift-lib-dir -l_swiftMockPlugin \
+// RUN:  -Wl,-rpath,%swift-lib-dir \
+// RUN:  -o %t/mock-plugin \
+// RUN:  %t/plugin.c
 
-// RUN: %swift-target-frontend -typecheck -verify -swift-version 5 -enable-experimental-feature Macros -load-plugin-executable %t/plugin#TestPlugin %t/test.swift
+// RUN: %swift-target-frontend \
+// RUN:   -typecheck -verify \
+// RUN:   -swift-version 5 -enable-experimental-feature Macros \
+// RUN:   -load-plugin-executable %t/mock-plugin#TestPlugin \
```

Comment:
Could maybe also have a substitution for this one that could take an optional name... not sure how useful it would be though.

Reply:
the binary name (`mock-plugin`)  and the module name (`TestPlugin`) are both test-file-specific.
I don't think we can reuse much.